### DISCUSSION TITLE
ghcup 0.1.15.1 supports Windows.

### DISCUSSION THREAD

**Discussion by kakkun61** [kakkun61/ghcman#18] [General]
ghcup 0.1.15.1 supports Windows.

https://gitlab.haskell.org/haskell/ghcup-hs/-/blob/master/CHANGELOG.md#01151-2021-06-11

Will ghcman (ghcups) be needed?

**Comment by jneira** [comment]
To be honest i cant see a strong argument to use ghcups instead ghcup now so i would say that it is not needed anymore.
I got no time to try ghcups but it was in my radar. Sure has been a nice alternative, so congrats.
I would encourage you to contribute to ghcup directly, your knowledge will be very welcomed and needed there.

**Reply by kakkun61** [reply]
Thank you for your honest opinion!

**Reply by jneira** [reply]
> To be honest i cant see a strong argument 

It does not suppose that argument does not exist of course. It would be great to know if you think there exist one of them.

**Comment by kakkun61** [comment]
I tried ghcup for Windows and I found:

- MSYS2 required
  - ghcman does not
- changing of GHC affects globally
  - using symbolic-link-like method
  - ghcman's effects are limited in local shells
    - changing the `Env:\PATH` shell variable
- command-line options are GNU style
  - ghcman has, of course, PowerShell style

I think that ghcup and ghcman have generally similar goals but that finely different goals.

**Reply by jneira** [reply]
Mmm those are convincing arguments. I would say that msys2 is required to build haskell packages which has a configure step (network for example). ghcup can install or use an existing msys2 installation. So i think it is appropiate to include optionally its intallation.
Agree with the other two arguments.
